.github/workflows/linux-compileonly.yaml

```
@@ -37,7 +37,11 @@ jobs:
           shopt -s dotglob && mv "$GITHUB_WORKSPACE"/* ${{ github.workspace }}
       - name: Install packages
         run: |
-          apt-get update && apt-get install -y cmake wget git
+          apt-get update && apt-get install -y wget git
+      - name: Install CMake
+        run: |
+          wget https://github.com/Kitware/CMake/releases/download/v3.23.5/cmake-3.23.5-linux-x86_64.tar.gz
+          sudo tar -C /usr/local --strip-components=1 -xf cmake-3.23.5-linux-x86_64.tar.gz
```

```suggestion
          tar -C /usr/local --strip-components=1 -xf cmake-3.23.5-linux-x86_64.tar.gz
```

Looks you are already root at this point.